client/state/sharing/keyring/reducer.js

```diff
@@ -28,12 +28,18 @@ export const isFetching = createReducer( false, {
 // Stores the list of available keyring connections
 export const items = createReducer( {}, {
 	[ KEYRING_CONNECTIONS_RECEIVE ]: ( state, { connections } ) => ( { ...keyBy( connections, 'ID' ) } ),
-	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: [ ...state[ ID ].sites, siteId.toString() ]
-	} } ),
-	[ PUBLICIZE_CONNECTION_DELETE ]: ( state, { ID, siteId } ) => ( { ...state, [ ID ]: {
-		...state[ ID ], sites: without( state[ ID ].sites, siteId.toString() )
-	} } ),
+	[ PUBLICIZE_CONNECTION_CREATE ]: ( state, { connection: { keyring_connection_ID: id, site_ID } } ) => {
```

Comment:
I think I mentioned it in the last pull request, but I think we should avoid destructuring more than a single level deep since it's difficult to read. Here, it's also inconsistent: we're aliasing `id` but not `site_ID` (would be nice to have it adhere to camelCase guidelines if not called as an object property), and we don't destructure `sites` yet it's referenced on the object in the function body anyways.